sqfvm-cpp/genericcmds.cpp

```diff
@@ -518,6 +518,111 @@ namespace
+	{
+		// create a copy of left array
+		auto result = std::make_shared<value>(left->as_vector());
+		result->data<arraydata>()->extend(right->as_vector());
+		return result;
+	}
+	std::shared_ptr<value> minus_array_array(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
+	{
+		// TODO: optimize (e.g. build a hash based set of the right array and check against it)
+		// TOOO: might require changes for checking array equality (https://github.com/X39/sqf-vm/issues/17)
+		// TODO: double check string equality
+		auto l = left->data<arraydata>();
+		auto r = right->data<arraydata>();
+		std::vector<std::shared_ptr<value>> result;
+		for (size_t i = 0; i < l->size(); i++)
+		{
+			auto current = l->at(i);
+			bool keep = true;
+			for (size_t j = 0; j < r->size(); j++)
+			{
+				auto check_against = r->at(j);
+				if (current->equals(check_against))
+				{
+					keep = false;
+					break;
+				}
+			}
+			if (keep)
+				result.insert(result.end(), current);
+		}
+		return std::make_shared<value>(result);
+	}
+	std::shared_ptr<value> append_array_array(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
+	{
+		left->data<arraydata>()->extend(right->as_vector());
+		return std::make_shared<value>();
+	}
+	std::shared_ptr<value> arrayintersect_array_array(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
+	{
+		// TODO: optimize (e.g. build a hash based ordered set)
+		// TOOO: might require changes for checking array equality (https://github.com/X39/sqf-vm/issues/17)
+		// TODO: double check string equality
+		auto l = left->data<arraydata>();
+		auto r = right->data<arraydata>();
+		std::vector<std::shared_ptr<value>> result;
+
+		for (size_t i = 0; i < l->size(); i++)
+		{
+			bool add = true;
+			auto current = l->at(i);
+			for (size_t j = 0; j < result.size(); j++)
+			{
+				auto check = result.at(j);
+				if (current->equals(check))
+				{
+					add = false;
+					break;
+				}
+			}
+
+			if (!add) continue; 
+
+			add = false;
+			for (size_t j = 0; j < r->size(); j++)
+			{
+				auto check = r->at(j);
+				if (current->equals(check))
+				{
+					add = true;
+					break;
+				}
+			}
+
+			if (add)
+				result.insert(result.end(), current);
+		}
+
+		return std::make_shared<value>(result);
+	}
+	std::shared_ptr<value> deleteat_array_scalar(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
+	{
+		auto l = left->data<arraydata>();
+		auto index = right->as_int();
+		if (index < 0 || index >= l->size())
+		{
+			vm->err() << "Array index out of bounds." << std::endl;
+			return std::make_shared<value>();
+		}
+		l->delete_at(index);
+		return std::make_shared<value>();
+	}
+	std::shared_ptr<value> find_array_any(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
+	{
+		// TODO: strings should be case sensitive
```

Comment:
using * sqf::value::dtype(void)` you can get the type and then explicitly check strings case-sensitive